File: workspace/apps/pidp/src/app/features/admin/party/party.model.ts
export interface PartyModel {
  username?: string;
  firstName?: string;
  lastName?: string;
  roles?: string[];
  email?: string;
  enabled: boolean;
  identityProvider?: string;
  created?: Date;
  participantId: number;
  keycloakUserId?: string;
  userIssues: Record<string, string>;
  systemsAccess: Record<string, SystemUserModel>;
}

export interface SystemUserModel {
  system: string;
  username: string;
  enabled: boolean;
  accountType: string;
  key: string;
  isAdmin: boolean;
  roles: string[];
  regions: string[];
}

File: backend/webapi/Features/Admin/PartyModel.cs
namespace Pidp.Features.Admin;

using NodaTime;

public class PartyModel
{
    public string Username { get; set; }
    public string FirstName { get; set; }
    public string LastName { get; set; }
    public List<string> Roles { get; set; } = new List<string>();
    public string Email { get; set; }
    public bool Enabled { get; set; }
    public string IdentityProvider { get; set; }
    public decimal ParticipantId { get; set; }
    public Instant? Created { get; set; }
    public Guid KeycloakUserId { get; set; }
    public Dictionary<string, string> UserIssues { get; set; } = new Dictionary<string, string>();
    public Dictionary<string, SystemUserModel> SystemsAccess { get; set; } = new Dictionary<string, SystemUserModel>();


}

public class SystemUserModel
{
    public string System { get; set; }
    public string Username { get; set; }
    public bool Enabled { get; set; }
    public string AccountType { get; set; }
    public string Key { get; set; }
    public bool IsAdmin { get; set; }
    public List<string> Roles { get; set; } = new List<string>();
    public List<string> Regions { get; set; } = new List<string>();

}
